File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.c
long long sumAndMultiply(int n) {
    long long digit;
    long long x= 0;
    long long place = 1;
    long long sum = 0;
    while(n != 0)
    {
        digit = n % 10;
        if(digit != 0)
        {
            x += digit * place;
            place *= 10;
            sum += digit;
        }
        n /= 10;
    }
    return x*sum;

}

File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution(object):
    def sumAndMultiply(self, n):
        """
        :type n: int
        :rtype: int
        """
        x = 0
        place = 1
        add = 0
        while(n):
            digit = n % 10
            if(digit != 0):
                x += digit * place
                add += digit
                place *= 10
            n = n // 10

        return x*add